modules/aleph/includes/aleph.reservation.inc

```diff
@@ -213,8 +215,57 @@ function aleph_reservation_list($account, $requested_type = NULL) {
   return $result;
 }
 
-function aleph_reservation_create() {
-  return array();
+/**
+ * Create a new reservation in the library system.
+ *
+ * @param object $account
+ *   Drupal user account object.
+ * @param string $id
+ *   ID of reservation.
+ *
+ * @return array
+ *   Reservation information.
+ *
+ * @throws DingProviderReservationExists
+ * @throws DingProviderReservationNotAllowed
+ * @throws DingProviderReservationNotAvailable
+ */
+function aleph_reservation_create($account, $id) {
+  $patron_handler = new AlephPatronHandler(aleph_client());
+  $patron_handler->setPatron(aleph_get_patron_by_account());
+
+  // Check if the users has this reservation and throw exception.
+  $reservation_exists = FALSE;
+  $reservations = $patron_handler->getReservations();
+
+  foreach ($reservations as $reservation) {
+    if ($id === $reservation->getItem()->getId()) {
+      $reservation_exists = TRUE;
+    }
+  }
+
+  if ($reservation_exists) {
+    throw new DingProviderReservationExists();
+  }
+
+  // Add the interest period for the 'to' date.
+  $interest_period = aleph_reservation_default_interest_period();
+
+  $end_request_date = new DateTime();
+  $end_request_date->add(new DateInterval('P' . $interest_period . 'D'));
+
+  // Build the reservation object.
+  $defaults = aleph_reservation_default_options($account);
+  $reservation = new AlephReservation();
+  $request = new AlephRequest();
+  $request->setRequestDate(new DateTime());
+  $request->setEndRequestDate($end_request_date);
+  $request->setPickupLocation($defaults['aleph_preferred_branch']);
+  $reservation->setRequest($request);
+
+  // Try to make the reservation.
+  $patron = $patron_handler->getPatron();
+  $patron_handler->createReservation($patron, $reservation);
```

Comment:
[Other](https://github.com/ding2/ding2/blob/master/modules/fbs/includes/fbs.reservation.inc#L117) [implementations](https://github.com/ding2/ding2/blob/master/modules/alma/includes/alma.reservation.inc#L304) of `reservation_create` have a return value. This does not. Is that intentional?

Reply:
As far as I can tell `createReservation` can throw a `\RuntimeException`. That does not seem to be declared here?

Reply:
Not necessarily a `\RuntimeException` however, you're right the return statement was missing.